src/test/java/StringManipulationTest.java

```diff
@@ -21,63 +20,133 @@ public void tearDown() {
         assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
     }
-
+    /**
+     *This is to test if the replacement of every nth term did take place with
+     * the space character
+     * */
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
     }
-
+    /**
+     *  this is to test for IndexOutOfBoundsException
+     * */
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
-    }
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+            manipulatedstring.removeNthCharacter(300, true);
+        });
 
+        String expectedMessage = "n must be less than the length of the string";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+    /**
+     *this is to test for IllegalArgumentException for negative n value
+     * */
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
-    }
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+            manipulatedstring.removeNthCharacter(-300, true);
+        });
+
+        String expectedMessage = "n must be greater than 0";
+        String actualMessage = exception.getMessage();
 
+        assertTrue(actualMessage.contains(expectedMessage));
+
+    }
+    /**
+     *this is to test for IllegalArgumentException for 0 as for n value
+     * */
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
-    }
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+            manipulatedstring.removeNthCharacter(0, true);
+        });
 
-    @Test
-    public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
-    }
+        String expectedMessage = "n must be greater than 0";
+        String actualMessage = exception.getMessage();
 
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+    /**
+     *  this is to test to handle null as just to return null of the string
+     *  data hasn't been setup yet. i could retrn "" but i chose null to show
+     *  null data type
+     * */
     @Test
-    public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+    public void testRemoveNthCharacter6() {
+        manipulatedstring.setString(null);
+        assertEquals(null, manipulatedstring.removeNthCharacter(3, true));
```

Comment:
By returning null here, you could introduce some unexpected behavior down the line because it isn't covered in the spec. Do you think it would make sense to just start the manipulator class as an empty string by default?